index.ts

```diff
@@ -173,18 +162,26 @@ class WertWidget {
         break;
     }
 
-    const customListener = this.listeners[event.data.type];
+    const customListener = this.listeners[event.data.type] as CustomListener<
```

Comment:
Do you think there's a way to go without typecasting here? I broke my brain while writing these types, they look good to me, but wouldn't work without it 🤔